tracer/src/Datadog.Trace/Ci/Net/TestOptimizationClient.GetSettingsAsync.cs

```diff
@@ -198,11 +200,14 @@ public readonly struct SettingsResponse
         [JsonProperty("default_branch")]
         public readonly string? DefaultBranch;
 
+        [JsonProperty("di_enabled")]
+        public readonly bool? DynamicInstrumentationEnabled;
```

Comment:
Is it intentionally nullable instead of giving it a default value (false)?

Reply:
Yes currently the TestOptimization product uses nullable everywhere to represent that it needs to gather information from the backend to make the final decision (true or false). Values can be rewritten using the environment variables (no nullable) and that prevents the backend request.

I actually don't like this and I'll do a refactor of all this mechanism in the future